### PR TITLE
Standardized rustdoc style

### PR DESCRIPTION
**Pull request by jamillambert**
Updated some of the rustdocs to make the style consistent.  There is still more to do which I can work on if it is useful?

### CODE REVIEW

**Review by tcharding — COMMENTED** [review]
Thanks man! Nice PR structure, correctly does one thing but manages to fix a bunch docs things at the same time. Good use review time.

One question and a bunch of full stops needed in the `crypto` module patch.

File: bitcoin/src/crypto/key.rs [hunk 1]
```diff
@@ -77,12 +77,12 @@ impl PublicKey {
         Ok(key.p2wpkh_script_code())
     }
 
-    /// Write the public key into a writer
+    /// Writes the public key into a writer
```
[the remark this answers is not in the document]
Comment (tcharding) [reply bitcoin/src/crypto/key.rs:80]: (You might want to use `git commit --amend --no-edit -a -n` to add the `crypto` module fixes to the last patch. You'll need to rebase (and force push) if you end up needing to change either of the other two.

File: units/src/amount.rs [hunk 2]
```diff
@@ -835,7 +835,7 @@ fn fmt_satoshi_in(
     Ok(())
 }
 
-/// Amount
+/// Amount.
```
[the remark this answers is not in the document]
Comment (storopoli) [reply units/src/amount.rs:838]: I think we shouldn't add periods to headings.

If you see technical docs
(for example, [RFC-6979: Deterministic Usage of the Digital Signature Algorithm (DSA) and Elliptic Curve Digital Signature Algorithm (ECDSA)](https://www.rfc-editor.org/rfc/rfc6979))
you'll see that the headings do not have periods.

[the remark this answers is not in the document]
Comment (storopoli) [reply units/src/amount.rs:838]: But they do "periodize" the item lists in several distinct forms:

- with the `;` followed by `.`  (2.3.1 first list)
- nothing (2.2)
- periods (several examples)

[the remark this answers is not in the document]
Comment (jamillambert) [reply units/src/amount.rs:838]: I agree we shouldn't add periods to headings since this does seem to be the standard, and in most places this is what has been done.  The exception is in headings for modules and structs e.g. below where the headings are shown as the description in the list and are inconsistent:
image

These headings are also not marked as a heading with a `#` and appear small on the page e.g. "A Bitcoin address.":

image


We could change them all with for example heading 2 `##` so it looks like below, note in the Struct list it is still displayed as standard text and not a heading style:
image


image

[the remark this answers is not in the document]
Comment (storopoli) [reply units/src/amount.rs:838]: I see, so we should pick either one of the options and standardize everything.
I understand your argument and see several valid points.
I don't have a preference, @apoelstra and @tcharding any opinions?

[the remark this answers is not in the document]
Comment (apoelstra) [reply units/src/amount.rs:838]: I took a quick glance at serde, which uses periods. We should also make sure we're using indefinite articles on each one (so "Amount" becomes "An amount."

We should not use `##` or any other heading. We had that a long time ago and somebody showed up here and lectured me about it, so I took them away. The idea is that these are not headings but a "summary description" of the types.

File: hashes/src/sha256t.rs [hunk 3]
```diff
@@ -141,7 +141,7 @@ macro_rules! sha256t_hash_newtype_tag {
     ($vis:vis, $tag:ident, $name:expr, $(#[$($attr:meta)*])*) => {
         #[doc = "The tag used for [`"]
         #[doc = $name]
-        #[doc = "`]\n\n"]
+        #[doc = "`].\n\n"]
```
Comment (storopoli) [inline hashes/src/sha256t.rs:144]: Can you explain to me why this change?

Comment (storopoli) [reply hashes/src/sha256t.rs:144]: Good! Thank you!

**Review by tcharding — APPROVED** [review]
ACK 5e7a638b4d7e2a0d87030046564f95e88b1a97c3

**Comment by tcharding** [comment]
Legend, thanks man! Keep at it.

**Review by apoelstra — APPROVED** [review]
ACK 5e7a638b4d7e2a0d87030046564f95e88b1a97c3
